pkg/utils/config-manager/main.go

```diff
@@ -0,0 +1,119 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/coreos/go-systemd/v22/dbus"
+	tomlv2 "github.com/pelletier/go-toml/v2"
+)
+
+const (
+	tomlFilePath = "/etc/containerd/config.toml"
+	nriPluginKey = "io.containerd.nri.v1.nri"
+	disableKey   = "disable"
+	replaceMode  = "replace"
+	resultDone   = "done"
+	unit         = "containerd.service"
+)
+
+func main() {
+	tomlMap, err := readConfig(tomlFilePath)
+	if err != nil {
+		log.Fatalf("Error reading TOML file: %v", err)
+	}
+
+	updatedTomlMap := updateNRIPlugin(tomlMap)
+
+	err = writeConfig(tomlFilePath, updatedTomlMap)
+	if err != nil {
+		log.Fatalf("failed to write updated config into a file %q:, %w", tomlFilePath, err)
+	}
+
+	err = restartSystemdUnit(unit)
+	if err != nil {
+		log.Fatalf("failed to restart containerd: %w", err)
+		os.Exit(1)
+	}
+}
+func writeConfig(file string, config map[string]interface{}) error {
+	var buf bytes.Buffer
+	enc := tomlv2.NewEncoder(&buf)
+	enc.SetIndentTables(true)
+	if err := enc.Encode(config); err != nil {
+		return fmt.Errorf("error encoding file: %w", err)
+	}
+
+	f, err := os.Create(file)
+	if err != nil {
+		log.Fatalf("error truncating the file: %q: %w", file, err)
```

Comment:
Why the different handling here then on the other error branches ?

Reply:
ah I missed to update this one, let me fix it quickly. Thanks for the catch

Reply:
fixed.